d2bs/kolbot/libs/common/Town.js

```diff
@@ -2168,10 +2168,8 @@ MainLoop:
 		}
 
 		if (act !== me.act) {
-			try {
-				Pather.useWaypoint(towns[act - 1], wpmenu);
-			} catch (WPError) {
-				throw new Error("Town.goToTown: Failed use WP");
+			if (!Pather.useWaypoint(towns[act - 1], wpmenu)) {
+				return false;
```

Comment:
yes, the same way `usePortal` does.
But if you don't have the wp at all, waiting will not fix the problem. Others players can't unlock wp for you.

Reply:
okay, so your change doesn't make sense
Pather.useWaypoint never returns false, it will either return true or throw an error which means your "return false" line will never be hit

if you are doing a script that needs a WP you don't have or can't get (since it tries to get it), then  you shouldn't be running that script

Reply:
Alright, didn’t look at useWaypoint.
As the discussion https://github.com/kolton/d2bot-with-kolbot/issues/1069
We thought that throwing error for not having access to a waypoint, is kind of too much restrictive.
For example, in mfhelper the leader may execute scripts not in ordered by act. The helper would change act but not have access to act via wp. But it would wait for the leader to come back in current act with another script, or stay in game until the end for exp boost or whatever.
Semantically, not having access to a wp is not an error, it is a use case that should be handled.

Reply:
I think it should stay as an error. There are post actions may rely upon taking that wp.
What you need to do is to handle the failed wp cases in MFHelper script explicitly.